clang/tools/driver/CMakeLists.txt

```diff
@@ -124,13 +124,7 @@ endif()
 set(features_file_src "${CMAKE_CURRENT_SOURCE_DIR}/features.json")
 set(features_file_dest "${CMAKE_BINARY_DIR}/share/clang/features.json")
 
-add_custom_command(OUTPUT ${features_file_dest}
-                   COMMAND ${CMAKE_COMMAND} -E make_directory
-                     ${CMAKE_BINARY_DIR}/share/clang
-                   COMMAND ${CMAKE_COMMAND} -E copy
-                     ${features_file_src}
-                     ${features_file_dest}
-                   DEPENDS ${features_file_src})
+configure_file(${features_file_src} ${features_file_dest})
```

Comment:
To be safer for the substitutions, I tend to use the `@ONLY` option so that we can mark the substitutions as `@NAME@` where `NAME` is a define that we control and specify the value using `add_definitions` or `set_soureces_properties`.

Reply:
I can put `@ONLY` in this PR.

We only need a variable in a downstream repo, so I'll try to make it work there.

Can you show me an example of `configure_file` using `add_definitions` and `set_source_files_properties`? The following doesn't work (using `@FOO` in the source file):

```
set(FOO "x")
configure_file(${features_file_src} ${features_file_dest} @ONLY)
set_source_files_properties(${features_file_src} PROPERTIES BAR "${FOO}")
```

Does this mechanism hide variables not explicitly set through `set_source_files_properties` or `add_definitions`?